src/NuGet.Clients/PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -166,24 +166,12 @@ public void ApplyShowPreviewSetting(bool show)
             UserSettings settings,
             Configuration.ISettings nugetSettings)
         {
-            if (settings == null)
-            {
-                if (nugetSettings == null)
-                {
-                    return;
-                }
-
-                // set depency behavior to the value from nugetSettings
-                SetSelectedDepencyBehavior(GetDependencyBehaviorFromConfig(nugetSettings));
-                return;
-            }
-
             _detailModel.Options.ShowPreviewWindow = settings.ShowPreviewWindow;
             _detailModel.Options.RemoveDependencies = settings.RemoveDependencies;
             _detailModel.Options.ForceRemove = settings.ForceRemove;
             _topPanel.CheckboxPrerelease.IsChecked = settings.IncludePrerelease;
 
-            SetSelectedDepencyBehavior(settings.DependencyBehavior);
+            SetSelectedDepencyBehavior(GetDependencyBehaviorFromConfig(nugetSettings)?? settings.DependencyBehavior);
```

Comment:
To make this easier to debug create a new var for the value before passing it into SetSelectedDepencyBehavior. There should be a space before ?? also.